File: helpdesk/recovery_tools/ExtractHistory.java
package vn.midomax.helpdesk;

import java.nio.file.*;
import java.util.*;
import java.io.*;
import java.util.regex.*;

public class ExtractHistory {
    public static void main(String[] args) throws Exception {
        String transcriptPath = "C:\\Users\\LENOVO\\.gemini\\antigravity-ide\\brain\\38583f59-4332-4d8b-b9ea-b09ce37171c8\\.system_generated\\logs\\transcript.jsonl";
        String outDir = "C:\\Users\\LENOVO\\.gemini\\antigravity-ide\\brain\\38583f59-4332-4d8b-b9ea-b09ce37171c8\\scratch\estore";
        Files.createDirectories(Paths.get(outDir));
        
        List<String> lines = Files.readAllLines(Paths.get(transcriptPath));
        Map<String, String> fileContents = new HashMap<>();
        
        for (String line : lines) {
            if (line.contains("\"created_at\"")) {
                Matcher m = Pattern.compile("\"created_at\":\"(.*?)\"").matcher(line);
                if (m.find()) {
                    String time = m.group(1);
                    if (time.compareTo("2026-06-28T13:00:00Z") > 0) {
                        break;
                    }
                }
            }
            
            if (line.contains("\"name\":\"write_to_file\"")) {
                Matcher targetM = Pattern.compile("\"TargetFile\":\"(.*?)\"").matcher(line);
                Matcher contentM = Pattern.compile("\"CodeContent\":\"(.*?)\",\"").matcher(line);
                if (targetM.find() && contentM.find()) {
                    String target = targetM.group(1).replace("\\\\", "\\");
                    String content = contentM.group(1).replace("\\n", "\n").replace("\\\"", "\"").replace("\\\\", "\\");
                    
                    String[] tracking = {"schedule.html", "dashboard.html", "admin-home.html", "user-home.html", "ticket-management.html", "ticket-modal-fragment.html", "HomeController.java", "TicketController.java", "fragments\\\\sidebar.html"};
                    for (String t : tracking) {
                        if (target.contains(t)) {
                            fileContents.put(t, content);
                        }
                    }
                }
            }
        }
        
        for (Map.Entry<String, String> entry : fileContents.entrySet()) {
            Files.write(Paths.get(outDir, entry.getKey().replace("\\\\", "_")), entry.getValue().getBytes());
            System.out.println("Restored: " + entry.getKey());
        }
    }
}

File: helpdesk/extract_history.py
import json
import os

transcript_path = r'C:\Users\LENOVO\.gemini\antigravity-ide\brain\38583f59-4332-4d8b-b9ea-b09ce37171c8\.system_generated\logs\transcript.jsonl'
output_dir = r'C:\Users\LENOVO\.gemini\antigravity-ide\brain\38583f59-4332-4d8b-b9ea-b09ce37171c8\scratch\restore'

if not os.path.exists(output_dir):
    os.makedirs(output_dir)

files_to_track = ['schedule.html', 'dashboard.html', 'admin-home.html', 'user-home.html', 'ticket-management.html', 'ticket-modal-fragment.html', 'HomeController.java', 'TicketController.java', 'sidebar.html']
file_contents = {}

with open(transcript_path, 'r', encoding='utf-8') as f:
    for line in f:
        try:
            entry = json.loads(line.strip())
            created_at = entry.get('created_at', '')
            
            # Stop if we are past 20:00 UTC+7 (which is 13:00:00Z)
            if created_at > "2026-06-28T13:30:00Z":
                break
                
            # Check tool_calls for write_to_file
            if 'tool_calls' in entry:
                for call in entry['tool_calls']:
                    if call['name'] == 'write_to_file':
                        target = call['args'].get('TargetFile', '')
                        content = call['args'].get('CodeContent', '')
                        for fname in files_to_track:
                            if fname in target:
                                file_contents[fname] = content
                                
            # Check RUN_COMMAND and view_file outputs
            if entry.get('type') in ['TOOL_RESPONSE', 'RUN_COMMAND']:
                content = entry.get('content', '')
                if '<!DOCTYPE html>' in content or 'package vn.midomax.helpdesk;' in content:
                    # heuristic to find which file it is
                    for fname in files_to_track:
                        if fname in content:
                            # Too risky to guess, but if it has a filename in the path before the content it might be safe
                            pass
        except Exception as e:
            pass

for fname, content in file_contents.items():
    with open(os.path.join(output_dir, fname), 'w', encoding='utf-8') as out:
        out.write(content)
    print(f'Restored {fname}')

print('Extraction complete!')
